### PR TITLE
BZ #2124469 automation

### PR DESCRIPTION
**Pull request by yosibsh**
Test PVC in 'bound' state after SubvolumeGroup deletion

Signed-off-by: Yosi Ben Shimon <ybenshim@redhat.com>

### CODE REVIEW

File: tests/manage/pv_services/test_pvc_delete_subvolumegroup.py [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    def test_pvc_delete_subvolumegroup(self, pvc_factory):
+        """
+        Delete PVC subvolumegroup and make sure PVC reaches bound state
+        """
+
+        logger.info("Creating pvc")
+        pvc_obj = pvc_factory(
+            interface=constants.CEPHFILESYSTEM,
+            size=5,
+            status=constants.STATUS_BOUND,
```
Comment (avd-sagare) [inline tests/manage/pv_services/test_pvc_delete_subvolumegroup.py:58]: Not needed. As its by default value

Comment (yosibsh) [reply tests/manage/pv_services/test_pvc_delete_subvolumegroup.py]: Done

File: tests/manage/pv_services/test_pvc_delete_subvolumegroup.py [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+            ceph_cmd="ceph fs subvolumegroup rm ocs-storagecluster-cephfilesystem csi",
+            format=None,
+        )
+
+        # List subvolumegroup (should be empty)
+        subvolume = self.tool_pod.exec_ceph_cmd(
+            ceph_cmd=self.subvolumegroup_ls_cmd, format=None
+        )
+        assert not subvolume, "Subvolumegroup should be empty"
+
```
Comment (PrasadDesala) [inline tests/manage/pv_services/test_pvc_delete_subvolumegroup.py:82]: Please add a subvolume group empty success message in the log, these info can be helpful when debugging the test failure

File: tests/manage/pv_services/test_pvc_delete_subvolumegroup.py [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+            size=5,
+            project=self.proj_obj,
+        )
+
+        # List subvolumegroup (should NOT be empty and contain 'csi')
+        subvolume = self.tool_pod.exec_ceph_cmd(
+            ceph_cmd=self.subvolumegroup_ls_cmd, format=None
+        )
+        assert subvolume, "Subvolumegroup should NOT be empty"
+        assert "csi" == subvolume[0]["name"], "Subvolumegroup should contain 'csi'"
```
Comment (PrasadDesala) [inline tests/manage/pv_services/test_pvc_delete_subvolumegroup.py:95]: Please add a success log message here

File: tests/manage/pv_services/test_pvc_delete_subvolumegroup.py [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+
+@skipif_ocs_version("<4.12")
+@polarion_id("OCS-4664")
+class TestPvcDeleteSubVolumeGroup(ManageTest):
+    """
+    Delete PVC subvolumegroup and make sure PVC reaches bound state
+    """
+
+    subvolumegroup_ls_cmd = (
+        "ceph fs subvolumegroup ls ocs-storagecluster-cephfilesystem"
```
Comment (PrasadDesala) [inline tests/manage/pv_services/test_pvc_delete_subvolumegroup.py:20]: This command may not work on an external mode cluster as the volume name would be different. 
I think it is `cephfs` please check and make the changes accordingly.

File: tests/manage/pv_services/test_pvc_delete_subvolumegroup.py [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+import logging
+from ocs_ci.ocs import constants
+from ocs_ci.ocs.resources.pvc import delete_pvcs
+from ocs_ci.ocs.resources.pod import get_ceph_tools_pod
+from ocs_ci.framework.testlib import ManageTest, skipif_ocs_version, polarion_id
+from ocs_ci.utility.utils import TimeoutSampler
+
+logger = logging.getLogger(__name__)
+
+
```
[the remark this answers is not in the document]
Comment (yosibsh) [reply tests/manage/pv_services/test_pvc_delete_subvolumegroup.py:18]: Done

File: tests/manage/pv_services/test_pvc_delete_subvolumegroup.py (hunk 2 above)
Comment (yosibsh) [reply tests/manage/pv_services/test_pvc_delete_subvolumegroup.py:104]: Done

File: tests/manage/pv_services/test_pvc_delete_subvolumegroup.py (hunk 3 above)
Comment (yosibsh) [reply tests/manage/pv_services/test_pvc_delete_subvolumegroup.py]: Done

File: tests/manage/pv_services/test_pvc_delete_subvolumegroup.py (hunk 4 above)
Comment (yosibsh) [reply tests/manage/pv_services/test_pvc_delete_subvolumegroup.py]: Done.

File: tests/manage/pv_services/test_pvc_delete_subvolumegroup.py [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+@skipif_ocs_version("<4.12")
+@polarion_id("OCS-4664")
+@bugzilla("2124469")
+@tier2
+class TestPvcDeleteSubVolumeGroup(ManageTest):
+    """
+    Delete PVC subvolumegroup and make sure PVC reaches bound state
+    """
+
+    vol_name = "ocs-storagecluster-cephfilesystem"
```
Comment (sidhant-agrawal) [inline tests/manage/pv_services/test_pvc_delete_subvolumegroup.py:28]: This is already present in defaults.py, so `defaults.CEPHFILESYSTEM_NAME` can be used here?

File: tests/manage/pv_services/test_pvc_delete_subvolumegroup.py [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+    external_mode = False
+    if config.DEPLOYMENT["external_mode"]:
+        vol_name = "cephfs"
```
Comment (sidhant-agrawal) [inline tests/manage/pv_services/test_pvc_delete_subvolumegroup.py:31]: Cephfs name may not  be same for every external cluster, I think better approach would be to fetch it from ENV_DATA or RHCS 
See this https://github.com/red-hat-storage/ocs-ci/blob/master/ocs_ci/deployment/helpers/external_cluster_helpers.py#L70

File: tests/manage/pv_services/test_pvc_delete_subvolumegroup.py [hunk 8]
```diff
@@ -0,0 +1,114 @@
...
+        external_mode = True
```
Comment (sidhant-agrawal) [inline tests/manage/pv_services/test_pvc_delete_subvolumegroup.py:32]: Is this `external_mode` being used somewhere and really needed?

File: tests/manage/pv_services/test_pvc_delete_subvolumegroup.py (hunk 6 above)
Comment (yosibsh) [reply tests/manage/pv_services/test_pvc_delete_subvolumegroup.py:28]: Not needed anymore.
Fetching the cephfs name from ENV_DATA or RHCS

File: tests/manage/pv_services/test_pvc_delete_subvolumegroup.py (hunk 7 above)
Comment (yosibsh) [reply tests/manage/pv_services/test_pvc_delete_subvolumegroup.py:31]: Done

File: tests/manage/pv_services/test_pvc_delete_subvolumegroup.py (hunk 8 above)
Comment (yosibsh) [reply tests/manage/pv_services/test_pvc_delete_subvolumegroup.py:32]: Removed

**Review by sidhant-agrawal — APPROVED** [review]
lgtm

**Review by PrasadDesala — APPROVED** [review]

**Pull request merged**